src/mlpack/methods/ann/augmented/tasks/copy_impl.hpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * @file copy_impl.hpp
+ * @author Konstantin Sidorov
+ *
+ * Implementation of CopyTask class
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_AUGMENTED_TASKS_COPY_IMPL_HPP
+#define MLPACK_METHODS_AUGMENTED_TASKS_COPY_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "copy.hpp"
+
+#include <cstdlib>
+#include <ctime>
+
+namespace mlpack {
+namespace ann /* Artificial Neural Network */ {
+namespace augmented /* Augmented neural network */ {
+namespace tasks /* Task utilities for augmented */ {
+
+CopyTask::CopyTask(size_t maxLength, size_t nRepeats) :
+    maxLength(maxLength),
+    nRepeats(nRepeats)
+{
+  assert(maxLength > 1);
+  // Just storing task-specific parameters.
+}
+
+void CopyTask::Generate(arma::field<arma::irowvec>& input,
+                            arma::field<arma::irowvec>& labels,
+                            size_t batchSize)
+{
+  input = arma::field<arma::irowvec>(batchSize);
+  labels = arma::field<arma::irowvec>(batchSize);
+  std::srand(unsigned(std::time(0)));
```

Comment:
I don't think it's a good idea to set a seed here, what if someone likes to generate the same output e.g. to debug some problem? If someone needs different results in a single run he can set ``math::RandomSeed(std::time(NULL));``before the run.